### DISCUSSION TITLE
How to convert the Javan Object in Scenario to JSON Objects

### DISCUSSION THREAD

**Discussion by sridharanss** [TouK/nussknacker#5528] [Q&A]
We are trying to use Nussknacker for building complete Scenario's, our requirement is to publish the inner object of a input kafka message to output kafka. 

The problem is, once I fetch the inner object using SPEL it is in Java format (with =), and if I pass that into Kafka it is not transformed as JSON instead it is going as Java object and getting error because of not able to pares.

**Comment by arkadius** [comment]
Can you provide the input message schema, output message schema and the spel expression extracting this inner object?
From what you've written it looks like somewhere a Record is converted to the string. It is a strange behaviour, the designer should report such a situation as a mistake. But to understand it better I need to know the wider context.

**Comment by sridharanss** [comment]
Yes, you are correct. We are passing the Record into string type filed of Kafaka payload as well as we are trying to build a JSON string by including the record.
Is there any inbuild mechanism available, to convert the Record to JSON object?

**Comment by arkadius** [comment]
Nussknacker do this automatically if the target type is also a record. It handles conversion between arvo records and json records and vice versa. But we don't have any mechanism allowing to convert a record to the json string. Probably it would be a good option to have such a feature in helpers - we already have a  `#CONV` helper. Helpers can be provided as an extension. We have a sample project demonstrating this: https://github.com/touk/nussknacker-sample-helpers but currently it is still not documented how to create such an extension and the API is not stable - we plan to cleanup it a bit.
